### PR TITLE
U1 login validations

### PR DESCRIPTION
**Pull request by lauuta185**
### Summary
In this PR, i added login validations and shared preferences to the application.

### Card
https://trello.com/c/FlfdupmJ/13-u1-maquetado-login-validaciones

### CODE REVIEW

File: app/src/main/java/com/example/wnews/HomeActivity.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import android.os.Bundle
+import android.widget.Toast
+
+class HomeActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.activity_home)
+        Toast.makeText(this, "Holaa!", Toast.LENGTH_LONG).show()
```
Comment (bdecorneliis) [inline app/src/main/java/com/example/wnews/HomeActivity.kt]: Remember to delete this line, was only for testing

File: app/src/main/java/com/example/wnews/MainActivity.kt [hunk 2]
```diff
@@ -2,10 +2,106 @@ package com.example.wnews
...
+        val view = binding.root
+        setContentView(view)
+
+        mainViewModel = ViewModelProvider(this)[MainViewModel::class.java]
+
+        binding.apply {
+
+
+            loginButton.setOnClickListener{
+                println("user:"+ loadData())
```
Comment (bdecorneliis) [inline app/src/main/java/com/example/wnews/MainActivity.kt]: Same as here

File: app/src/main/java/com/example/wnews/MainActivity.kt [hunk 3]
```diff
@@ -2,10 +2,106 @@ package com.example.wnews
...
+                val firstNameInput = firstName.editableText.toString()
+                val secondNameInput= secondName.editableText.toString()
+
+                mainViewModel.setLogin(firstNameInput, secondNameInput)
+
+                if(loginValidation(firstNameInput, secondNameInput)){
+                    // fun save userData()
```
Comment (bdecorneliis) [inline app/src/main/java/com/example/wnews/MainActivity.kt]: We strongly recommend not leaving commented lines

File: app/src/main/java/com/example/wnews/MainActivity.kt [hunk 4]
```diff
@@ -2,10 +2,106 @@ package com.example.wnews
...
+                    val intent = Intent(this@MainActivity, HomeActivity::class.java)
+
+                    startActivity(intent)
+                }
+            }
+            fun initHome(){
+                loginButton.setOnClickListener { mainViewModel.setLogin(firstName =String(), secondName =String()) }
```
Comment (bdecorneliis) [inline app/src/main/java/com/example/wnews/MainActivity.kt]: Why you put "= String()" here?

File: app/src/main/java/com/example/wnews/MainActivity.kt [hunk 5]
```diff
@@ -2,10 +2,106 @@ package com.example.wnews
...
+    fun setObserve(){
+        mainViewModel.newLoginModel.observe(this@MainActivity) { model->
+            println(model.firstName)
+        }
+    }
+
+
+    private val SHARED_PREFS = "sharedPrefs"
+    private val KEY_EMAIL = "keyEmail"
+    private val PASSWORD = "password"
```
Comment (bdecorneliis) [inline app/src/main/java/com/example/wnews/MainActivity.kt]: Leave this variables in "companion object" at the bottom of the document:
companion object{
private const val SHARED_PREFS = "sharedPrefs"
...
}

File: app/src/main/java/com/example/wnews/MainActivity.kt [hunk 6]
```diff
@@ -2,10 +2,106 @@ package com.example.wnews
...
+            return false
+        }
+
+        if (firstNameInput.isNotEmpty() && !firstNameInput.contains("@")) {
+            binding.firstName.error = "Please, enter a valid email."
+            return false
+        }
+
+        if(secondNameInput.isEmpty()) {
+            binding.secondName.error = "This field cannot be empty."
```
Comment (bdecorneliis) [inline app/src/main/java/com/example/wnews/MainActivity.kt]: Once we talk about leave plain text in the document. This kind of text, have to be in "string.xml" file. And then, you call it like this:
 binding.firstName.error = getString(R.string.error_empty)

File: app/src/main/res/layout/activity_home.xml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+    <Button
+        android:id="@+id/Logout"
+        android:text="Logout"
+        android:layout_width="300dp"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="64dp"
+        android:layout_marginTop="340dp"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
```
Comment (bdecorneliis) [inline app/src/main/res/layout/activity_home.xml:17]: Same here, sizes into dimen.xml and text into string.xml

**Review by bdecorneliis — CHANGES_REQUESTED** [review]
Few comments to improve.
You are beginning to understand logic and Kotlin. Let's keep going!

**Review by LisandroDiMeo — CHANGES_REQUESTED** [review]
agree with bruno suggestions. Also, I recommend that after you declare a variable, method, or whatever, think on a more declarative name (like MainActivity could be renamed to LoginActivity, and MainViewModel could be renamed to LoginViewModel, and so on). Even though this isn't a functional change, stopping the ball and thinking on what are you doing while creating things will give you a better picture of what are you doing (and what you shouldn't ;) )

**Review by bdecorneliis — APPROVED** [review]
Amazing:rocket::rocket::rocket:

**Pull request merged**